br/pkg/lightning/restore/precheck_impl.go

```diff
@@ -153,13 +162,23 @@ func (ci *clusterResourceCheckItem) Check(ctx context.Context) (*CheckResult, er
 		return nil, errors.Trace(err)
 	}
 	estimateSize := clusterSource * replicaCount
-	if estimateSize > clusterAvail {
+	if estimateSize <= tikvAvail && estimateSize <= tiflashAvail {
+		theResult.Message = fmt.Sprintf("The storage space is rich, which TiKV is %s and TiFlash is %s. The estimated storage space is %s.",
+			units.BytesSize(float64(tikvAvail)), units.BytesSize(float64(tiflashAvail)), units.BytesSize(float64(estimateSize)))
+	}
+
+	if estimateSize > tikvAvail {
 		theResult.Passed = false
-		theResult.Message = fmt.Sprintf("Cluster doesn't have enough space, available is %s, but we need %s",
-			units.BytesSize(float64(clusterAvail)), units.BytesSize(float64(estimateSize)))
-	} else {
-		theResult.Message = fmt.Sprintf("Cluster available is rich, available is %s, we need %s",
-			units.BytesSize(float64(clusterAvail)), units.BytesSize(float64(estimateSize)))
+		theResult.Message += fmt.Sprintf("The estimated storage space required by the TiKV cluster is %s but the actual storage space is %s.",
+			units.BytesSize(float64(tikvAvail)), units.BytesSize(float64(estimateSize)))
+	}
+	if estimateSize > tiflashAvail {
```

Comment:
What happens if a table has no tiflash replica or a cluster has no tiflash nodes?

Reply:
thanks for your review, I will add these check

Reply:
add it in https://github.com/pingcap/tidb/commit/263bba7236e98d724bedbc692c4c85508ac309e0 and https://github.com/pingcap/tidb/commit/a898ff616020d515ab4a48313c17058091b4994f. ptal~